src/meta/src/hummock/manager/checkpoint.rs

```diff
@@ -190,7 +191,10 @@ impl HummockManager {
         let versioning = versioning_guard.deref_mut();
         assert!(new_checkpoint.version.id >= versioning.checkpoint.version.id);
         versioning.checkpoint = new_checkpoint;
-        versioning.mark_objects_for_deletion();
+        // Not delete stale objects when archive is enabled
+        if !self.env.opts.enable_hummock_data_archive {
+            versioning.mark_objects_for_deletion();
```

Comment:
If not mark for deletion, the size of stale_objects will keep increasing until full GC. The size of stale_objects can be very large because full GC is infrequent.